src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue29921.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue29921 : _IssuesUITest
+{
+	public Issue29921(TestDevice testDevice) : base(testDevice)
+	{
+	}
+	public override string Issue => "Flyout icon not replaced after root page change";
+
+	[Test]
+	[Category(UITestCategories.FlyoutPage)]
+	public void FlyoutIconUpdatedAfterInsertPageBefore()
+	{
+		App.WaitForElement("InsertPageButton");
+		App.Tap("InsertPageButton");
+		VerifyScreenshot();
```

Comment:
Pending snapshot on Mac and Windows.
![image](https://github.com/user-attachments/assets/395fa522-0b97-43f6-b746-6eaab00c9b3c)
Could you commit the images?

Reply:
@jsuarezruiz I have updated the pending snapshots for Mac and Windows.